### PR TITLE
Secretize cleanup

### PR DESCRIPTION
**Pull request by AlexanderViand-Intel**
Addresses the first two points of #1107 plus some other small cleanup around Secretize.

* `--secretize` now applies to all functions in the module, unless the (renamed) `function=<name>` flag is given, allowing the flag to be omitted most of the time.
* `--mlir-to-secret-arithmetic` (and the `-to-ckks/-to-bgv` pipelines that use it) no longer includes `--secretize` (or its flags) and instead, test files now either include `--secretize` or (preferred) explicit `{secret.secret}` annotations. 
Note that the `to-openfhe` versions still have an `entry-function` flag, as that seems to be required for some of the openFHE stuff.
I did _not_ yet remove secretize from the boolean pipelines since I'm less familiar with them, but I think that would be the right approach there, too (even if we still need to designate an entry function)  with an eye towards making the pipelines more "user/frontend-friendly" rather than "testing-friendly".

And some other stuff thrown in:
* `secretize.cpp` lives in `lib/Transforms/secretize` but the tests lived in `lib/Dialect/Secret/Transforms/secretize`. I moved the tests since that's easier, but we could also consider moving the pass implementation instead.
* `--wrap-generic` now uses the recent [walk-based pattern rewriter](https://github.com/llvm/llvm-project/pull/113825). I'm sure there are other one-shot passes like this that don't need the fixed-point iteration logic of the greedy rewriter.
* I used Hongren's new alias magic to clean up `naive_matmul.mlir`  by simply roundtripping it through heir-opt. Again, I'm sure there's more cleanup opportunities like this we can watch out for as we touch other parts of the code/tests.

### CODE REVIEW

**Comment by j2kun** [comment]
> remove secretize from the boolean pipelines

This sounds fine to me, at least for consistency's sake.

File: tests/Examples/openfhe/naive_matmul.mlir
Comment (j2kun) [inline tests/Examples/openfhe/naive_matmul.mlir:1]: So much cleaner 😅

**Comment by j2kun** [comment]
Well done. Approving and marking as pull_ready now, unless you want to also remove secretize from the boolean pipelines in this same PR.

**Review by j2kun — APPROVED** [review]

**Comment by AlexanderViand-Intel** [comment]
> Well done. Approving and marking as pull_ready now, unless you want to also remove secretize from the boolean pipelines in this same PR.

Squashed :) I'll deal with the boolean stuff another time (or hope @asraa  tackles it instead xD)

**Review by asraa — APPROVED** [review]
Thanks! It's true that secretize is more of a testing convenience artifact.

**Pull request merged**